encoding/mvt/unmarshal.go

```diff
@@ -32,7 +32,12 @@ func UnmarshalGzipped(data []byte) (Layers, error) {
 // Unmarshal takes Mapbox Vector Tile (MVT) data and converts into a
 // set of layers, It does not project the coordinates.
 func Unmarshal(data []byte) (Layers, error) {
-	return unmarshalTile(data)
+	layers, err := unmarshalTile(data)
+	if err != nil && dataIsGZipped(data) {
+		return nil, errors.New("Failed to unmarshal the tile, data seems to be compressed in GZip format, try to use the UnmarshalGzipped method from this package")
```

Comment:
it might be better to define the error up top and return that. Something like
```
var ErrDataIsGZipped = errors.New("failed to unmarshal, data possibly gzipped")
```
then you can do this easier if you want
```
_, err := Unmarshal(...)
if err == ErrDataIsGZipped {
  UnmarshalGzipped(...)
}

Reply:
Thanks for your feedback, being a new Go programmer (I've learned the language the last year but I started to use it at work just few months ago) I really benefit from it